src/main/python/org/broadinstitute/hellbender/tools/gcnv/README.txt

```diff
@@ -0,0 +1,43 @@
+==============================
+GATK gCNV computational kernel
+==============================
+
+`gcnvkernel` is a python module that provides the computational backend for GATK
+germline copy number variant (gCNV) tools and workflows.
+
+This module implements inference schemes for read-depth profile denoising, germline
+integer copy number variation discovery, germline contig ploidy determination, associated
+I/O methods, and helper CLI scripts. `gcnvkernel` additionally provides general-purpose
+inference schemas built on the top of `PyMC3` and `theano`.
+
+The module is organized as follows::
+
+    `gcnvkernel.inference`: Classes related to the general task of variational inference,
+        such as convergence tracking, deterministic annealing protocol, and structured
+        stochastic optimizers.
+
+    `gcnvkernel.io`: I/O routines for saving and loading various data structures, such as
+        genomic intervals, read count data, global and sample-specific posteriors, and
+        sample metadata.
+
+    `gcnvkernel.models`: `PyMC3` model declarations, `theano` symbolic operations (e.g.
+        forward-backward algorithm for HMMs), and custom probability distributions.
+
+    `gcnvkernel.preprocess`: Routines for filtering interval lists.
+
+    `gcnvkernel.structs`: Classes that represents genomic intervals, interval annotations,
+        and sample metadata.
+
+    `gcnvkernel.tasks`: Inference task executors for different models and their use cases.
+        This module additionally provides general-purpose templates for setting up arbitrary
+        ADVI inference tasks for models with mixed discrete and continuous latent variables.
+
+    `gcnvkernel.utils`: Contains utility classes for CLI tools, as well as an implementation
+        of recursive least squares (RLS).
+
+Even though `gcnvkernel` can be used as a standalone tool, its intended use case is to be
+privately called by GATK CLI tools. As such, the outputs of `gcnvkernel` are considered
```

Comment:
I think @vdauwera and @droazen need to know about this gcnvkernal tool if they do not already. Currently, it is not something we document and having this readme in the repo will ultimately lead to questions from the users. I mean, I have questions--namely, what is this and how do I use it?

Reply:
I'm pretty sure @droazen is aware of it, but @vdauwera might not be.  `gcnvkernel` is the name of the python package that provides the machine-learning backend for the new gCNV tools (DetermineGermlineContigPloidy and GermlineCNVCaller).  It relies heavily on the pymc3 package (https://github.com/pymc-devs/pymc3) to specify and perform inference for the gCNV models.  DetermineGermlineContigPloidy and GermlineCNVCaller are essentially Java wrappers that simply pass arguments to the respective python scripts in the gcnvkernel package; one might imagine that advanced users could directly use the python scripts, so gcnvkernel is essentially standalone.  This is not fundamentally different from how some tools in the GATK are simply wrappers of various R scripts, it's just that the thing being wrapped is sufficiently complex to warrant the development of an entire standalone package.

Reply:
Would I invoke the gcnvkernal tool much like we do CatVariants?

- CatVariants doc: https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_gatk_tools_CatVariants.php

If so, gcnvkernal will need its own tool doc, like CatVariants.

Reply:
No, a typical user would simply use the DetermineGermlineContigPloidy and GermlineCNVCaller tools, which are already documented.